### DISCUSSION TITLE
Tag jumping decontamination

### DISCUSSION THREAD

**Discussion by prototaxites** [Shenhav-and-Korem-labs/SCRuB#17] [General]
Hi, 

We recently got a 16S rRNA amplicon dataset back that is heavily contaminated by tag jumping that occurred during the library  preparation - the gist of the issue is that sequences from the same ASV but with different tags can form heteroduplexes in the amplicon pool, following which polymerase activity during adapter ligation can cause the tags to switch. As we are using combinatorial indexing due to our large number of samples, this can cause sequences to move from one sample to another, causing cross-sample contamination. We have some ability to estimate this rate, however, because this process also causes tags to switch into unused tag combinations - of 576 combinations, we have only used approximately 380, so any sequences in the remaining combinations can only be caused by this process.

I have been looking at how to deal with this problem - my first instinct was to incorporate this into a Bayesian occupancy model ([see here for full details](https://discourse.mc-stan.org/t/appropriate-count-distribution-for-false-detection-in-occupancy-models/32986?u=prototaxites)), where we model the number of reads in a sample as the sum of contaminant and non-contaminant reads with Negtive binimial likelihoods, and using the number of reads in an unused combination to estimate that constant contamination rate. However, the model does not seem to be particularly identifiable as currently constructed.

I then remembered about this package, and looking at the documentation the underlying model seems quite similar to mine above - although my approach was to try and estimate everything in one step rather than to decontaminate the data and then do further downstream analysis. 

1. Would SCRuB be appropriate for removing this sort of contamination? The contamination is not spatial, and these unused combinations do not represent true samples - they are an artefact of the library preparation process.
2. If yes, can the package handle overdispersed contamination? I've had a shot at modelling these contaminant reads directly, and they are quite strongly overdispersed - I am concerned that samples with high levels of contamination would not be fully decontaminated if overdispersion is not accounted for.

Many thanks!

**Comment by talkorem** [comment]
Hi there,

Thanks for posting this - a really interesting setting. 
This is akin to well-to-well leakage with a prescribed (potential) pattern. SCRuB does have a way of handling this phenomenon but only as it relates to leakage into negative controls, but currently not when it pertains to leakage between samples. The first (handling just controls) should be a rather simple adaptation - requires editing N_d from "Addressing well-to-well leakage" in the SCRuB manuscript - but I really don't think that's what you want. We had some ideas for future developments related to the second; If you want to collaborate on something related to this we're happy to look into it. Is there a good benchmark dataset for this problem?

As for overdispersion - I'm not sure how you define overdispersion with respect to SCRuB's model. We've been able to successfully model high levels of quite noisy contamination. 

Best,
Tal
